R/ResamplingCustomCV.R

```diff
@@ -46,14 +46,21 @@ ResamplingCustomCV = R6Class("ResamplingCustomCV", inherit = Resampling,
     #' @param task [Task]\cr
     #'   Used to extract row ids.
     #'
-    #' @param f (`factor()`)\cr
+    #' @param split (`factor()`)\cr
     #'   Row ids are split on this factor, each factor level results in a fold.
     #'   Empty factor levels are dropped and row ids corresponding to missing values are removed,
     #'   c.f. [split()].
-    instantiate = function(task, f) {
+    instantiate = function(task, split) {
       task = assert_task(as_task(task))
-      assert_factor(f, len = task$nrow, all.missing = FALSE)
-      self$instance = split(task$row_ids, f, drop = TRUE)
+      if (is.factor(split)) {
+        assert_factor(split, empty.levels.ok = FALSE, len = task$nrow, all.missing = FALSE)
+      } else if (is.character(split)) {
```

Comment:
This is undocumented and not clear to me. If this is a character, you want to split on the columns referenced by this vector? The code in the block does not seem to work if there are multiple columns provided. I also wonder if it would be safer to provide two separate arguments here (mutually exclusive).

Reply:
> If this is a character, you want to split on the columns referenced by this vector? The code in the block does not seem to work if there are multiple columns provided. 

Yes, when `split` is chr I would like to split by this existing vector within the task.

> I also wonder if it would be safer to provide two separate arguments here (mutually exclusive).

No strong opinion on that, would make documentation more distinct but also might raise confusion which argument should be used for which effect.

I've documented it and added another assertion. If this still feels wrong to you, feel free to change it to your liking :)